### DISCUSSION TITLE
leaderboard problems

### DISCUSSION THREAD

**Discussion by PMorghK** [ppy/osu#20284] [Q&A] [answered]
![Screenshot_20220913-013100](https://user-images.githubusercontent.com/43874888/189817013-40409dea-6389-472f-9d39-48297de9da50.png)

As you can see my score is highr but I am not on the leaderboards.

Simple as that.

**Comment by peppy** [comment]
please provide any kind of detail. a screenshot cannot help us in any way. link the beatmap, include your replay. read the discussion tips and include logs.

hel pdeveloper

**Reply by xstarrystar** [reply]
I noticed this problem like 2 days ago.
My score is 582,346 and it's top 903 in the leaderboard, but it should be top 7 according to the leaderboard currently showing up, the current top 7 score is 530,000.

Leaderboard shows **correctly** when on the menu after finishing the map: 
![image](https://user-images.githubusercontent.com/61137868/189820297-4015cf7d-12dd-4b86-9f5c-c458c8c2403e.png)

Shows **incorrectly** when on beatmap selection: 
![image](https://user-images.githubusercontent.com/61137868/189820870-a3ea9578-fec2-4e07-a2ac-38a6496625e8.png)

beatmap: https://osu.ppy.sh/beatmapsets/1073074#osu/2245774

Replay: https://cdn.discordapp.com/attachments/910628317898612746/1019122834968170547/PADORU_Sotarks_Gift.osr (wasn't able to upload here)

Log files:
[network.log](https://github.com/ppy/osu/files/9553891/network.log)
[performance.log](https://github.com/ppy/osu/files/9553892/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9553893/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9553894/updater.log)
[database.log](https://github.com/ppy/osu/files/9553895/database.log)

**Reply by peppy** [reply]
@smoogipoo can you take a look into this? you'll probably be able to identify the issue faster than i can.

**Reply by smoogipoo** [reply]
The problem here appears to be scores which were set with the old mod multipliers. The top 50 on that map actually has >1.1M score stored internally, which is the value used by osu!web. The client recalculates scores when they're displayed so it's fooling you - maybe it shouldn't be doing this anymore.

In https://github.com/ppy/osu-queue-score-statistics/pull/87 I implemented a way to reprocess scores which will fix this.

**Comment by Non-Binary-Programmer** [comment]
I found a similar thing in the leaderboard for [Garven's Insane diff of Remote Control](https://osu.ppy.sh/beatmapsets/53857#osu/164020).
Video (showing 50th NM score which is much higher than 50th overall score):

https://user-images.githubusercontent.com/100527514/204150278-fda24bd6-2c96-4d44-afe9-170be16bc3ac.mp4

Additionally, the gameplay leaderboard is also incorrect:
<img width="335" alt="in-game leaderboard showing incorrect score" src="https://user-images.githubusercontent.com/100527514/204150882-8dcc1b1a-4587-445a-b717-2456a61a04aa.png">

Logs:
[database.log](https://github.com/ppy/osu/files/10098816/database.log)
[input.log](https://github.com/ppy/osu/files/10098817/input.log)
[network.log](https://github.com/ppy/osu/files/10098818/network.log)
[performance-draw.log](https://github.com/ppy/osu/files/10098819/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/10098820/performance-update.log)
[performance.log](https://github.com/ppy/osu/files/10098821/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10098822/runtime.log)

**Comment by smoogipoo** [comment, marked as answer]
The issue with these is that the osu-queue-score-statistics upgrade failed last time. This will be fixed by https://github.com/ppy/osu-queue-score-statistics/pull/117 + the subsequent recalc.
